### PR TITLE
mjson: Initial include of package

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR adds a package for [mjson](https://github.com/cesanta/mjson/)

### Testing procedure

A small test application is provided

### Issues/PRs references

None

### CODE REVIEW

**Review by aabadie — COMMENTED** [review]
Maybe add an `app.config.test` to check Kconfig  (and add Kconfig dependency modelling) ?

**Comment by bergzand** [comment]
> Maybe add an `app.config.test` to check Kconfig (and add Kconfig dependency modelling) ?

Added an `app.config.test` file. Not sure if there is more Kconfig dependency to model here.

**Comment by bergzand** [comment]
> Since Kconfig is on it's way out, I don't think we need to worry too much about that.

Are you suggesting to leave out the Kconfig integration here? Or just the `app.config.test` file?

File: pkg/mjson/mjson.mk [hunk 1]
```diff
@@ -0,0 +1,5 @@
+MODULE = mjson
+
+CFLAGS += -DMJSON_ENABLE_MERGE=0
```
[the remark this answers is not in the document]
Comment (benpicco) [reply pkg/mjson/mjson.mk:5]: We actually *do* provide it (well, newlib/picolibc does) but it's a bad idea to use it with our small stacks.

**Review by benpicco — APPROVED** [review]

**Comment by benpicco** [comment]
I'm just saying don't put too much work into Kconfig when we are going to delete it anyway by the end of the month.

File: pkg/mjson/mjson.mk [hunk 2]
```diff
@@ -0,0 +1,7 @@
+MODULE = mjson
+
+# The mjson_merge function requires an alloca implementation, disabled to
+# prevent heap memory allocations
```
Comment (benpicco) [inline pkg/mjson/mjson.mk:4]:
```suggestion
# prevent stack memory allocations
```

Comment (bergzand) [reply pkg/mjson/mjson.mk]: Fixed

**Comment by benpicco** [comment]
CI is not yet happy :confused:

**Comment by MrKevinWeiss** [comment]
Looks like you either need to add an `rsource` in the `pkg/Kconfig` to bring in the module or remove kconfig and the app.config.test to disable the test.

**Comment by bergzand** [comment]
> Looks like you either need to add an `rsource` in the `pkg/Kconfig` to bring in the module

Thanks, completely missed that bit!

**Comment by bergzand** [comment]
Thanks for adding the final bits @MrKevinWeiss :heart:

**Pull request merged**